src/prefect/runner/submit.py

```diff
@@ -94,62 +120,85 @@ async def _submit_flow_to_runner(
+        flow_run_id = uuid.UUID(response.json()["flow_run_id"])
 
-        return uuid.UUID(flow_run_id)
+        return await client.read_flow_run(flow_run_id)
 
 
 @sync_compatible
 async def submit_to_runner(
     prefect_callable: Union[Flow, Task],
-    parameters: Dict[str, Any],
-) -> uuid.UUID:
+    parameters: Union[Dict[str, Any], List[Dict[str, Any]]],
+    retry_failed_submissions: bool = True,
+) -> Union[uuid.UUID, List[uuid.UUID]]:
     """
-    Run a callable in the background via the runner webserver.
+    Submit a callable in the background via the runner webserver one or more times.
 
     Args:
         prefect_callable: the callable to run (only flows are supported for now, but eventually tasks)
-        parameters: the keyword arguments to pass to the callable,
-    """
-
-    flow_run_id = await _submit_flow_to_runner(prefect_callable, parameters)
-
-    if inspect.isawaitable(flow_run_id):
-        return await flow_run_id
-    else:
-        return flow_run_id
-
-
-@sync_compatible
-async def submit_many_to_runner(
-    prefect_callable: Union[Flow, Task], parameters_list: List[Dict[str, Any]]
-) -> List[uuid.UUID]:
-    """
-    Run multiple callables in the background via the runner webserver, respecting the run limit.
-
-    Args:
-        prefect_callable: the callable to run (flows or tasks)
-        parameters_list: a list of dictionaries, each representing keyword arguments to pass to the callable.
+        parameters: keyword arguments to pass to the callable. May be a list of dictionaries where
+            each dictionary represents a discrete invocation of the callable
+        retry_failed_submissions: Whether to retry failed submissions to the runner webserver.
     """
+    if not PREFECT_EXPERIMENTAL_ENABLE_EXTRA_RUNNER_ENDPOINTS.value():
+        raise ValueError(
+            "The `submit_to_runner` utility requires the `Runner` webserver to be"
+            " built with extra endpoints enabled. To enable this, set the"
+            " `PREFECT_EXPERIMENTAL_ENABLE_EXTRA_RUNNER_ENDPOINTS` setting to `True`."
+        )
 
-    n_current_runs = await get_current_run_count()
-    available_slots = max(0, PREFECT_RUNNER_PROCESS_LIMIT.value() - n_current_runs)
+    if isinstance(parameters, dict):
+        parameters = [parameters]
 
     submitted_run_ids = []
-    for parameters in parameters_list[:available_slots]:
-        flow_run_id = await submit_to_runner(prefect_callable, parameters)
-        if inspect.isawaitable(flow_run_id):
-            flow_run_id = await flow_run_id
-        submitted_run_ids.append(flow_run_id)
+    unsubmitted_parameters = []
+
+    for p in parameters:
+        try:
+            flow_run_id = await _submit_flow_to_runner(
+                prefect_callable, p, retry_failed_submissions
+            )
+            if inspect.isawaitable(flow_run_id):
+                flow_run_id = await flow_run_id
+            submitted_run_ids.append(flow_run_id)
```

Comment:
nit: I have a habit of trying to keep my exception blocks as tight as possible - what do you think of moving this response data processing into an `else:` clause attached to this try/except?

Reply:
good call - added in {some commit i cant find for some reason but its here}